core/src/main/java/org/elasticsearch/snapshots/Snapshot.java

```diff
@@ -277,9 +293,14 @@ public static Snapshot fromXContent(XContentParser parser) throws IOException {
         int totalShard = 0;
         int successfulShards = 0;
         List<SnapshotShardFailure> shardFailures = NO_FAILURES;
-
-        XContentParser.Token token = parser.currentToken();
-        if (token == XContentParser.Token.START_OBJECT) {
+        if (parser.currentToken() == null) { // fresh parser? move to the first token
+            parser.nextToken();
+        }
+        if (parser.currentToken() == XContentParser.Token.START_OBJECT) {  // on a start object move to next token
+            parser.nextToken();
+        }
+        XContentParser.Token token;
+        if ((token = parser.nextToken()) == XContentParser.Token.START_OBJECT) {
```

Comment:
I don't quite understand this logic, we are looking for a double-nested object? like `{ { ... } }`?